squad-server/index.js

```diff
@@ -353,16 +336,26 @@ export default class SquadServer extends EventEmitter {
     await this.logParser.watch();
   }
 
+  // DEPRECATED FUNCTION
   getAdminPermsBySteamID(steamID) {
-    return this.admins[steamID];
+    return this.getAdminPermsByAnyID(steamID);
+  }
+
+  getAdminPermsByAnyID(anyID) {
+    // using this.players directly to keep the function sync
+    const player = anyIDToPlayer(anyID, this.players);
+    if (player === undefined) return;
+    for (const parm in player)
+      if (parm.endsWith("ID") && player[parm] in this.admins)
+        return this.admins[player[parm]];
   }
 
   getAdminsWithPermission(perm) {
     const ret = [];
-    for (const [steamID, perms] of Object.entries(this.admins)) {
-      if (perm in perms) ret.push(steamID);
+    for (const [anyID, perms] of Object.entries(this.admins)) {
+      if (perm in perms) ret.push(anyID);
     }
-    return ret;
+    return anyIDsToPlayers(ret, this.players).map(player => player.eosID);
```

Comment:
As discussed on Discord with @Ulibos, the best choice appears to be a method that accepts an extra parameter to tune the output based on the needs.

A jsdoc would be a good fit for defining all the possible values for the extra parameter that I will just refer to as "output format parameter".

The output format parameter could be defined with the following jsdoc: `@param {'steamID'|'eosID'|'any'|'player'}` and will return the output formatted as showed in the following example:

- **steamID**: The output stays the same as the current implementation. Example: `[ '01234567891234567', '01234567891234567' ]`
- **eosID**: Example: `[ 'a1b2c3d4e5f6a7b8c9d1e2f3a4b5c6d7e8f9', 'a1b2c3d4e5f6a7b8c9d1e2f3a4b5c6d7e8f9' ]`
- **any**: The eosID should be the preferred ID but include a fallback to steamID in case of missing eosID. Example `[ 'a1b2c3d4e5f6a7b8c9d1e2f3a4b5c6d7e8f9', '01234567891234567' ]`
- **player**: `[ { name: 'example_name', eosID: 'a1b2c3d4e5f6a7b8c9d1e2f3a4b5c6d7e8f9', steamID: '01234567891234567', ... }, { name: 'example_name', eosID: 'a1b2c3d4e5f6a7b8c9d1e2f3a4b5c6d7e8f9', steamID: null|undefined, ... } ]`